### PR TITLE
Spring cross-context calls with interface

### PR DESCRIPTION
**Pull request by yuanyuancin**
1. add samples for spring cross-context calls
2. add docs for spring cross-context calls
3. optimize spring cross-context calls with interface

### CODE REVIEW

File: docs/content/zh-cn/docs/tutorials/module-development/module-and-base-communication.md [hunk 1]
```diff
@@ -3,6 +3,168 @@ title: 模块与模块、模块与基座通信
 weight: 300
 ---
 
+基座与模块之间、模块与模块之间存在 spring 上下文隔离，互相的 bean 不会冲突、不可见。然而很多应用场景比如中台模式、独立模块模式等存在基座调用模块、模块调用基座、模块与模块互相调用的场景。
+
+# Spring 环境
+
+## 基座调用模块
+
```
Comment (lvjing2) [inline docs/content/zh-cn/docs/tutorials/module-development/module-and-base-communication.md:11]: 补充几句说明

File: sofa-serverless-runtime/sofa-serverless-common/src/main/java/com/alipay/sofa/serverless/common/service/ServiceProxyFactory.java [hunk 2]
```diff
@@ -127,30 +138,29 @@ private static Object doCreateServiceProxy(Biz biz, Object service, ClassLoader
...
 
         SpringServiceInvoker serviceInvoker = new SpringServiceInvoker(service, biz.getBizName(),
             biz.getBizVersion(), biz.getIdentity(), clientClassLoader, service
                 .getClass().getClassLoader());
         ProxyFactory factory = new ProxyFactory();
-        Class<?> targetClass = service.getClass();
-        if (targetClass.isInterface()) {
-            factory.addInterface(targetClass);
+        if (serviceType.isInterface()) {
+            factory.addInterface(serviceType);
```
Comment (lvjing2) [inline sofa-serverless-runtime/sofa-serverless-common/src/main/java/com/alipay/sofa/serverless/common/service/ServiceProxyFactory.java:153]: 单测需要覆盖

File: sofa-serverless-runtime/arklet-core/src/main/resources/com/alipay/sofa/arklet/log/log4j/log-conf.xml
[the remark this answers is not in the document]
Comment (yuanyuancin) [reply sofa-serverless-runtime/arklet-core/src/main/resources/com/alipay/sofa/arklet/log/log4j/log-conf.xml:1]: 我印象中我根本没改过这个？emmmm，maybe是执行啥命令自动改的？我改回去

**Review by lvjing2 — APPROVED** [review]
LGTM

**Pull request merged**